src/main/kotlin/com/project/hotel/configuration/RouterConfiguration.kt

```diff
@@ -0,0 +1,17 @@
+package com.project.hotel.configuration
+
+import com.project.hotel.handler.UserHandler
+import kotlinx.coroutines.FlowPreview
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.web.reactive.function.server.coRouter
+
+@Configuration
+class RouterConfiguration {
+    @FlowPreview
+    @Bean
+    fun productRoutes(productsHandler: UserHandler) = coRouter {
+        POST("/", productsHandler::addUser)
```

Comment:
да это дефолтный пример реализации coRouter
но поробуй сделать не  через прокидования в функцию а через лямбду 

```suggestion
        POST("/") { request ->
              var userCommand = request.awaitBody()
              productsHandler. addUser(userCommand)
        }
```
это такой некий подход когда ты описываешь Router как Controller
тоесть оставляешь в этой функции все что связано с получением данных и респонсом
а логику обработки и перечисления выносишь в другой класс

Reply:
Добавлю, что иногда стоит вынести обработчик в отдельную функцию, если на него есть ссылка по другому пути (но вообще если один и тот же обработчик по разным путям это не оч хорошо). Или если он содержит очень большой кусок кода, но тут его можно вынести в конец, чтобы он не мешал.
И когда выносишь в отдельный метод, то лучше (то что я видел  в большинстве случаев и как мне кажется) писать реализовать как метод в этом же классе, а не делать зависимость от еще одного, который по сути выполняет ту же задачу.